ckcc/protocol.py

```diff
@@ -259,52 +259,62 @@ class CCProtocolUnpacker:
     # - given full rx message to work from
     # - this is done after un-framing
 
-    @classmethod
-    def decode(cls, msg):
+    @staticmethod
+    def decode(msg):
         assert len(msg) >= 4
         sign = str(msg[0:4], 'utf8', 'ignore')
 
-        d = getattr(cls, sign, cls)
-        if d is cls:
+        d = getattr(CCProtocolUnpacker, sign, None)
+        if d is None:
             raise CCFramingError('Unknown response signature: ' + repr(sign))
 
         return d(msg)
-        
 
     # struct info for each response
-    
+
+    @staticmethod
     def okay(msg):
         # trivial response, w/ no content
         assert len(msg) == 4
         return None
 
     # low-level errors
+    @staticmethod
     def fram(msg):
-        raise CCFramingError("Framing Error", str(msg[4:], 'utf8'))
+        raise CCFramingError("Framing Error: " + str(msg[4:], 'utf8'))
```

Comment:
this line is the actual fix --> `CCProtoError` wants just msg as an argument

Reply:
otherwise just added `@staticmethod` decorator so it is clear that `msg` is not `self`